js/modules/k6/html/html.go

```diff
@@ -44,17 +47,201 @@ type Selection struct {
 	sel *goquery.Selection
 }
 
-func (s Selection) Add(arg goja.Value) Selection {
-	switch val := arg.Export().(type) {
+func (s Selection) emptySelection() Selection {
+	// Ask for out of bounds item for an empty selection.
+	return s.Eq(s.Size())
+}
+
+func (s Selection) buildMatcher(v goja.Value, gojaFn goja.Callable) func(int, *goquery.Selection) bool {
+	return func(idx int, sel *goquery.Selection) bool {
+		fnRes, fnErr := gojaFn(v, s.rt.ToValue(idx), s.rt.ToValue(sel))
+
+		if fnErr != nil {
+			panic(fnErr)
+		}
+
+		return fnRes.ToBoolean()
+	}
+}
+
+func (s Selection) varargFnCall(arg interface{},
```

Comment:
You're already doing a type switch, so why not just add another case for `goja.Value` that calls itself again with the exported value?